src/statusbar.c

```diff
@@ -273,10 +273,10 @@ static void Statusbar_OverlayInit(const SDL_Surface *surf)
 	OverlayLedRect.x = surf->w - 5*h/2;
 	OverlayLedRect.y = h/2;
 	/* free previous restore surface if it's incompatible */
-	if (OverlayUnderside &&
-	    OverlayUnderside->w == OverlayLedRect.w &&
-	    OverlayUnderside->h == OverlayLedRect.h &&
-	    OverlayUnderside->format->BitsPerPixel == surf->format->BitsPerPixel)
+	if (OverlayUnderside && (
+	    OverlayUnderside->w != OverlayLedRect.w &&
```

Comment:
Shouldn't that rather be || at the end instead of && ? (same for the following line)

Reply:
Thank you. I made that mistake when I was transcribing my solution. Fixed now.